generators/csharp/sdk/src/endpoint/http/HttpEndpointGenerator.ts

```diff
@@ -312,4 +309,347 @@ export class HttpEndpointGenerator extends AbstractEndpointGenerator {
+        const snippet = this.getHttpPagerMethodSnippet({ endpoint });
+        return csharp.method({
+            name: this.context.getEndpointMethodName(endpoint),
+            access: csharp.Access.Public,
+            isAsync: false,
+            parameters,
+            summary: endpoint.docs,
+            return_,
+            body: csharp.codeblock((writer) => {
+                const requestParam = endpointSignatureInfo.requestParameter;
+                if (!requestParam) {
+                    throw new Error("Request parameter is required for pagination");
+                }
+                const unpagedEndpointMethodName = this.context.getEndpointMethodName(endpoint);
+                const unpagedEndpointResponseType = getEndpointReturnType({ context: this.context, endpoint });
+                if (!unpagedEndpointResponseType) {
+                    throw new Error("No return type found for endpoint");
+                }
+
+                endpoint.pagination?._visit({
+                    offset: (pagination) => {
+                        const offsetType = this.context.csharpTypeMapper.convert({
+                            reference: pagination.page.property.valueType
+                        });
+                        // use specified type or fallback to int
+                        const stepType = pagination.step
+                            ? this.context.csharpTypeMapper.convert({
+                                  reference: pagination.step?.property.valueType
+                              })
+                            : csharp.Type.object();
+                        const offsetPagerClassReference = this.context.getOffsetPagerClassReference({
+                            requestType: requestParam.type,
+                            requestOptionsType,
+                            responseType: unpagedEndpointResponseType,
+                            offsetType,
+                            stepType,
+                            itemType
+                        });
+                        writer.write("var pager = new ");
+                        writer.writeNode(offsetPagerClassReference);
+                        writer.write("(");
+                        writer.indent();
+                        writer.writeLine(`${requestParam.name},`);
+                        writer.writeLine(`${optionsParamName},`);
+                        writer.writeLine(`${unpagedEndpointMethodName},`);
+                        writer.writeLine(`request => ${this.nullableDotGet("request", pagination.page)} ?? 0,`);
+
+                        writer.writeLine("(request, offset) => {");
+                        writer.indent();
+                        this.initializeNestedObjects(writer, "request", pagination.page);
+                        writer.writeTextStatement(`${this.dotGet("request", pagination.page)} = offset`);
+                        writer.dedent();
+                        writer.writeLine("},");
```

Comment:
Is it possible to use `csharp.instantiateClass` for this? We're otherwise repeating the logic required to write every individual parameter.